### PR TITLE
DOC: fix read_excel and ExcelFile engine parameter description (#28231)

### LINKED ISSUE #28231: Update documentation on pd.read_excel, to reflect the fact that support for OpenDocument files is available
```python
class ExcelFile:
    """
    Class for parsing tabular excel sheets into DataFrame objects.
    Uses xlrd. See read_excel for more documentation

    Parameters
    ----------
    io : string, path object (pathlib.Path or py._path.local.LocalPath),
        file-like object or xlrd workbook
        If a string or path object, expected to be a path to xls or xlsx file.
    engine : string, default None
        If io is not a buffer or path, this must be set to identify io.
        Acceptable values are None or ``xlrd``.
    """

    from pandas.io.excel._odfreader import _ODFReader
    from pandas.io.excel._openpyxl import _OpenpyxlReader
    from pandas.io.excel._xlrd import _XlrdReader

    _engines = {"xlrd": _XlrdReader, "openpyxl": _OpenpyxlReader, "odf": _ODFReader}

```
#### Problem description

I was searching for support for reading _.ods_ files, and upon reading the documentation for _read_excel_ and IO related functions, I found nothing. 

When I was about to search for another library, I discovered that support had already been added to the _read_excel_ function (as of #2311), but the docs weren't updated, either in _ExcelFile_, or in the _read_excel_ function, and as suggested when creating the issue https://pandas-docs.github.io/pandas-docs-travis/ also wasn't. I also didn't find any related issue regarding this.

I might be missing something, since it is my first issue, sorry about that if it's the case.

I'd suggest even something along the lines of: 
```
    Parameters
    ----------
    io : string, path object (pathlib.Path or py._path.local.LocalPath),
        file-like object or xlrd workbook
        If a string or path object, expected to be a path to xls or xlsx file.
    engine : string, default None
        If io is not a buffer or path, this must be set to identify io.
        Acceptable values are None, ``xlrd``, ``openpyxl`` or ``odf``.
        Note that ``odf`` reads tables out of OpenDocument formatted files.
    """
```
for both _ExcelFile_ and _read_excel_. That would make visible that support for other engines is available.

Thanks.

### PR DESCRIPTION
**Pull request by zys5945**
Closes #28231.

First PR. Not sure if I'm doing anything wrong.

### CODE REVIEW

File: pandas/io/excel/_base.py [hunk 1]
```diff
@@ -787,7 +787,8 @@ class ExcelFile:
         If a string or path object, expected to be a path to xls or xlsx file.
```
Comment (WillAyd) [inline pandas/io/excel/_base.py]: The short summary of this docstring still says that it uses `xlrd` - can you update that while touching these?

**Review by WillAyd — CHANGES_REQUESTED** [review]

File: pandas/io/excel/_base.py (hunk 1 above)
Comment (zys5945) [reply pandas/io/excel/_base.py]: Thank you for pointing it out. Just pushed a change.

**Review by WillAyd — APPROVED** [review]
lgtm - ping on green

**Comment by zys5945** [comment]
@WillAyd tests passed

**Comment by WillAyd** [comment]
Thanks @zys5945

**Pull request merged**